### ISSUE TITLE
Support Advanced JSON object address building for mapquest

### ISSUE THREAD

**Issue by alexkavon** [nchaulet/node-geocoder#143]
The following syntax does not work for using "mapquest" as the service:
~~~
geocoder.geocode({address: '29 champs elysée', country: 'France', zipcode: '75008'}, function(err, res) {
    console.log(res);
});
~~~

Instead it utilizes the basic example. While this seems to work pretty well, it would be nice to build out addresses as a JSON object. MapQuest allows for fields such as city, state, postalCode, etc. See usage examples for fields as well as full list here: http://www.mapquestapi.com/geocoding/#locations

**Comment by nchaulet** [comment]
it could be a great feature :+1:

**Comment by nchaulet** [comment]
Just published v3.6.0 who supports Advanced JSON object address building for mapquest

**Comment by alexkavon** [comment]
:D
